File: backends/python-fastapi/app/usecase/notification/dto.py
"""
通知ユースケース Dto モジュール。

Author: Satoshi Nagashiba <satoshi.nagashiba@gmail.com>
"""
from dataclasses import dataclass, field
from typing import Optional


@dataclass
class NotificationStoreDto:
    """通知登録 Dto。"""

    title: str
    body: Optional[str] = None


@dataclass
class NotificationBulkReadDto:
    """通知一括既読 Dto。"""

    executor_id: int
    ids: list[int] = field(default_factory=list)
    all_flag: bool = False


@dataclass
class NotificationPatchDto:
    """通知更新 Dto。"""

    notification_id: int
    read: Optional[bool] = None
    title: Optional[str] = None
    message: Optional[str] = None

File: backends/ts-hono/src/usecase/notification/dto.ts
/**
 * 通知ユースケース DTO モジュール。
 *
 * @author Satoshi Nagashiba <satoshi.nagashiba@gmail.com>
 */

/** 通知一括既読入力 DTO。 */
export interface NotificationBulkReadInput {
  executorId: number;
  ids: number[];
  allFlag: boolean;
}

/** 通知ファンアウト入力 DTO。 */
export interface NotificationFanOutInput {
  title: string;
  body?: string;
}

/** 通知部分更新入力 DTO。 */
export interface NotificationPatchInput {
  id: number;
  read?: boolean;
  title?: string;
  message?: string;
}
